kotlinx-coroutines-core/common/src/JobSupport.kt

```diff
@@ -594,17 +619,27 @@ public open class JobSupport constructor(active: Boolean) : Job, ChildJob, Paren
         cancelImpl(parentJob)
     }
 
-    // Child was cancelled with cause
-    // It is overridden in supervisor implementations to ignore child cancellation
-    public open fun childCancelled(cause: Throwable): Boolean =
-        cancelImpl(cause) && handlesException
+    /**
+     * Returns `true` if job should cancel itself on child [CancellationException].
+     */
+    public open fun cancelOnChildCancellation(cause: CancellationException) = false
```

Comment:
It seems wasteful to have a separate overridable function only for the use by flow scopes. I suggest to have them override `childCancelled` just supervisor jobs do.